Expensify/App
# Migrate BankAccount_SetupWithdrawalAccount API

### Details
I tried to keep things familiar here while migrating stuff over from Web-Secure. That means for now, I'm sticking with a single page `ReimbursementAccountPage` + then conditionally showing views based on the `achData.currentStep` so basically it's very similar to what we have in `Web-Secure`.

There are some changes that I am proposing that feel more inline with how we are doing things in `Expensify.cash`

- Got rid of the "magic" methods used in `Web-Secure` used to move between steps, call the withdrawal account API, validate forms, etc. Mostly I found that pattern to be pretty difficult to reason about and the convenience isn't worth the confusion (that's just my opinion, but we don't have any patterns like this in E.cash so far).
- Instead, every view will handle it's own validation, form state, submit action, and navigation via the header in some cases so `nextStep()` and `previousStep()` are no longer triggered in the parent view (`addReimbursementPage.jsx` in `Web-Secure`) and instead implemented by each "step". This might be a little more verbose, but I think it's better to be explicit.
- The country selector in the `BankAccountStep` is not included since all the accounts added for the free plan should be `US`.
- Instead of having a `this.achData` to reference we are referencing + storing the data for an account in setup in the Onyx cache / `localStorage`. Most of the logic is left intact, but everywhere we refer to `this.achData` is now will refer to Onyx data. This is ideal because it eliminates the need for something like [`ReimbursementAccountPage.refreshView()`](https://github.com/Expensify/Web-Secure/blob/044c82affb78812a58b881a6d5ba026d91dace3b/site/app/dialogs/reimbursementAccount/reimbursementAccountPage.jsx#L450-L450). Instead we can navigate to the next step simply by setting the Onyx data and letting the view which is subscribing to this data update itself in response.

### Fixed Issues (**Part 2**)
https://github.com/Expensify/Expensify/issues/166041

### Tests
1. Navigate to `/add-verified-bank-account`
2. Verify the "Add Bank Account" view appears
3. Select "Connect Manually"
4. Enter the test credentials:

>Routing Number: 011401533
>Account Number: 1111222233331111

5. Press Save & Continue button
6. Verify you land on the "Company Information" step
7. Check the Chrome dev tools Network tab and verify that `BankAccount_SetupWithdrawalAccount` request was made (Web Only)
7. Press the "<" button in the top left of the header
8. Verify you are back on the selection screen
9. Select "Log Into Your Bank"
10. Enter the Plaid test credentials

>Login: `user_good`
>Password: `pass_good`
11. Follow prompts until you reach the account select screen
12. Select the savings account
13. Tap Save & Continue button
14. Verify you land in the "Company Information" step and the API request was successful

### QA Steps
No QA necessary until the e2e flow is ready for testing

### Tested On


### Screenshots


#### Web
![2021-06-11_12-37-38](https://user-images.githubusercontent.com/32969087/121755402-cb602f80-cab2-11eb-8e76-f4a053e965a1.png)
![2021-06-11_12-39-45](https://user-images.githubusercontent.com/32969087/121755405-cd29f300-cab2-11eb-9b7a-3182e903b518.png)
![2021-06-11_12-39-53](https://user-images.githubusercontent.com/32969087/121755409-ce5b2000-cab2-11eb-9e17-3df729b7bde4.png)
![2021-06-11_12-39-53](https://user-images.githubusercontent.com/32969087/121755412-d024e380-cab2-11eb-97de-d39f920cdd57.png)
![2021-06-11_12-41-50](https://user-images.githubusercontent.com/32969087/121755416-d1eea700-cab2-11eb-97c7-d5a8d300eda2.png)
![2021-06-11_12-42-47](https://user-images.githubusercontent.com/32969087/121755420-d3b86a80-cab2-11eb-9220-33dc9f559435.png)
![2021-06-11_12-42-55](https://user-images.githubusercontent.com/32969087/121755423-d4e99780-cab2-11eb-98ac-5ba70bad1eea.png)
![2021-06-11_12-43-07](https://user-images.githubusercontent.com/32969087/121755425-d74bf180-cab2-11eb-83f3-aa57dc15805f.png)
![2021-06-11_12-43-14](https://user-images.githubusercontent.com/32969087/121755431-d915b500-cab2-11eb-8935-9bd0b885a13f.png)

Not including other screenshot because no UI is being added here. The web flow is provided as a reference.
